src/hooks/internal/useNavigation.ts

```diff
@@ -26,11 +30,61 @@ export type UseNavigationReturnType = {
         label: string | undefined
         navigateTo: () => void
     }
+    months: Array<{
+        label: string
+        value: number
+    }>
+    navigateToMonth: (month: number) => void
+    navigateToYear: (year: number) => void
 }
+
+type Month = {
+    value: number
+    dateForLabel?: Temporal.PlainDate
+}
+
+const getAvailableMonths = (calendarSystem = 'iso8601'): Month[] => {
```

Comment:
let's pre-calculate the available months for each calendar for all the calendars we have, and save that data as JSON resource .. and only _try_ to calculate on the fly if a calendarSystem that's not pre-loaded is passed .. I think that will be a much more efficient way of doing this code, it's a lot of calls to Temporal now for something that doesn't change per calendar, inside a loop, that will probably keep being re-rendered by the UI component 

do you think that would work?

(only the localisation, in that case, can be outside the preloaded data)

Reply:
You mean something similar to [this](https://github.com/dhis2/multi-calendar-dates/blob/c86d95f3999b509ae036eb33ccdd9903dc0357d6/src/custom-calendars/calendarLocalisations.ts#L18) ?

Reply:
> You mean something similar to [this](https://github.com/dhis2/multi-calendar-dates/blob/c86d95f3999b509ae036eb33ccdd9903dc0357d6/src/custom-calendars/calendarLocalisations.ts#L18) ?

yes